src/node.rs

```diff
@@ -48,28 +76,183 @@ impl Builder {
         self
     }
 
-    pub fn port(mut self, port: u16) -> Self {
-        self.port = port;
+    pub fn at_directory(mut self, directory: impl Into<Directory>) -> Self {
+        self.directory = directory.into();
         self
     }
 
-    /// It supposed to open node at the supplied location
-    pub async fn build(self, directory: impl Into<Directory>) -> Result<Node, ()> {
-        let _ = directory;
-        todo!()
+    /// Start a node with supplied parameters
+    pub async fn build(self) -> anyhow::Result<Node> {
+        let Self {
+            mode: Mode::Full,
+            chain,
+            directory,
+            name,
+        } = self;
+
+        let chain_spec = match chain {
+            Chain::Gemini2a => ConsensusChainSpec::from_json_bytes(
+                include_bytes!("../res/chain-spec-raw-gemini-2a.json").as_ref(),
```

Comment:
nitpick: do we need to [guard the relative path](https://doc.rust-lang.org/std/fs/fn.canonicalize.html) usage here and/or maybe grab the absolute path?

On a separate note, how we are going to handle when [chain specs are updated upstream](https://github.com/subspace/subspace/tree/main/crates/subspace-node/res)?

Reply:
Also linking @i1i1 [PR here regards to chainspec management](https://github.com/subspace/subspace/pull/893)

Reply:
This thing happens during the compilation, so no need to guard the relative path. Anyway, this code will be removed in next pr.

Reply:
Chain spec is only updated when we launch a new chain. After that it doesn't change (well, we may edit bootstrap nodes if we need, but that is about it).